l10n_br_fiscal/views/document_fiscal_line_mixin_view.xml

```diff
@@ -40,7 +41,10 @@
                                 attrs="{'invisible': [('tax_icms_or_issqn', '!=', 'issqn')]}"
                             >
                 <group>
-                  <field name="issqn_tax_id" />
+                  <field
+                                        name="issqn_tax_id"
+                                        attrs="{'readonly': [('rel_tax_calc', '=', 'tax_calc_manual')]}"
```

Comment:
Os "readonly" incluídos dentro desse arquivo estão corretos ? Não seria '!=' ao invês de '=' assim [('rel_tax_calc', '!=', 'tax_calc_manual')] , pelo o que entendi o campo só pode ser alterado na tela no caso de ser tax_calc_manual.

Reply:
Nesse caso, os que eu coloquei como "readonly" são os campos select que permitem selecionar os impostos que foram cadastrados previamente no sistema. Se for possível selecionar algum deles, entra no caso do "semi-automático", o que foge um pouco do manual em que eles teriam que preencher os demais campos.

Reply:
Não sei se entendi bem @ygcarvalh , esses campos poderão ser selecionados na tela tanto no caso Manual quando no Semi-Automatico seria isso ? Porque então não seria o caso de fazer attrs="{'readonly': [('rel_tax_calc', '=', 'tax_calc_auto')]}" quer dizer o campo é readonly apenas no caso do Calculo do Imposto ser Automático?

Reply:
ah sim @mbcosta eu tive o mesmo questionamento, mas veja bem esta correcto: imagina que vc edita a aliquota do ISSQN manualmente, dai vc nao deve poder selecionar o m2o issqn_tax_id, certo? Portanto eu acho que isso ta OK.

Reply:
Esses campos poderiam ser selecionados tanto no caso Automático quanto no caso Semi-Automático. O Automático seria justamente o comportamento padrão do fiscal: quando selecionamos um produto, todos os impostos são preenchidos automaticamente e ficam readonly, exceto esses campos casos o usuário tenha desejo de alterar alguma regulamentação. Por exemplo: Large Cabinet tem, por padrão, um ICMS de 12% no sistema, mas se eu quero colocar como sendo de 18%, eu posso fazer isso selecionando esse valor no ICMS Tax ID;

No caso do Semi-Automático, ao selecionar o produto, os campos não são preenchidos, e seria papel do usuário selecionar qual o Tax ID que ele deseja para, então, o sistema realizar os cálculos. Pegando o exemplo anterior, quando eu selecionar o produto Large Cabinet, esses campos vão continuar vazios e eu posso selecionar algum desses valores (ICMS Tax ID, IPI Tax ID, ...) e, então, os demais campos são preenchidos pelo sistema;

Por fim, no Manual seria o contexto do usuário preencher todos os campos por conta e risco. Por isso não teria tanto sentido se ele fosse capaz de selecionar alguma regulamentação que esteja cadastrada no sistema, porque teria o mesmo comportamento do Semi-Automático. Por exemplo, eu selecionei o Pedal Bin como produto e a Operação está setada como Manual, eu teria que preencher qual o valor base, a percentagem e o valor calculado do ICMS, do IPI, e dos outros impostos na mão, meio que fazendo isso numa calculadora/papel e depois jogando esses valores calculados nos respectivos campos.